docs/source/chainermn/model_parallel/model_parallel_on_chainermn.rst

```diff
@@ -0,0 +1,202 @@
+Model Parallel on ChainerMN
+===========================
+
+
+Step 1: Communicators
+~~~~~~~~~~~~~~~~~~~~~
+
+To perform multi-node communications, a *communicator* is needed.
+Basic usages are the same with the case of the data parallel, see :doc:`../tutorial/step1_communicators_optimizers`::
+
+    comm = chainermn.create_communicator()
+
+If you want to define collective communications among limited number of processes later, it is useful to split the communicator::
+
+    subcomm = comm.split(comm.rank % 2, comm.rank)
+
+.. figure:: ../../../image/model_parallel/comm_split.png
+    :align: center
+    :scale: 50%
+
+For further detail about the communicator split, please refer to `MPI tutorial <http://mpitutorial.com/tutorials/introduction-to-groups-and-communicators/>`__.
+
+
+
+Step 2: Datasets and Iterators
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In model parallel, there are three types of processes regarding to datasets.
+
+1. model inputs come from datasets, and each process takes different mini-batches
+2. model inputs come from datasets, and several processes share the same mini-batches
+3. model inputs come from other processes
+
+
+1. scatter_dataset
+------------------
+
+For the first case, you may use ``scatter_dataset`` as is introduced in :doc:`../tutorial/step2_datasets_evaluators`.
+
+.. figure:: ../../../image/model_parallel/scatter_dataset.png
+    :align: center
+
+2. multi node iterator
+----------------------
+
+For the second case, iterator need to be modified, where ``create_multi_node_iterator`` is useful::
+
+    train, test = chainer.datasets.get_mnist()
+    train_iter = chainermn.iterators.create_multi_node_iterator(
+        chainer.iterators.SerialIterator(train, batchsize), comm)
+    test_iter = chainermn.iterators.create_multi_node_iterator(
+        chainer.iterators.SerialIterator(test, batchsize), comm)
+
+The resulting iterators return the same mini-batches among processes specified by the communicator.
+
+.. figure:: ../../../image/model_parallel/multi_node_iterator.png
+    :align: center
+
+3. empty dataset
+----------------
+
+For the last case, you may use ``create_empty_dataset``, which returns a dataset with the same number of empty tuples as the original dataset::
```

Comment:
I think examples must be shown, referring to example 1 or MultiNodeNStepLSTM, as "empty" dataset is not intuitive and more followup is needed to make the reason clearer.